packages/p2p/stores/buy-sell-store.js

```diff
@@ -0,0 +1,360 @@
+        } else {
+            const response = await requestWS({ p2p_order_info: 1, id: order.p2p_order_create.id });
+            this.form_props.handleConfirm(response.p2p_order_info);
+            this.form_props.handleClose();
+        }
+
+        setSubmitting(false);
+    };
+
+    @action.bound
+    hideAdvertiserPage() {
+        this.setShowAdvertiserPage(false);
+    }
+
+    @action.bound
+    hideVerification() {
+        this.setShouldShowVerification(false);
+    }
+
+    @action.bound
+    loadMoreItems() {
+        this.setItemOffset(0);
+
+        requestWS({
+            p2p_advert_list: 1,
+            counterparty_type: this.table_type,
+            offset: this.item_offset,
+            limit: this.general_store.list_item_limit,
+        }).then(response => {
+            if (!response.error) {
+                const { list } = response.p2p_advert_list;
+                this.setHasMoreItemsToLoad(list.length >= this.general_store.list_item_limit);
+                this.setItems(list);
+                this.setItemOffset((this.item_offset += list.length));
+            } else {
+                this.setApiErrorMessage(response.error.message);
+            }
+            this.setIsLoading(false);
+        });
+    }
+
+    @action.bound
+    onCancelClick() {
+        this.setShouldShowPopup(false);
+    }
+
+    @action.bound
+    onChangeTableType(event) {
+        this.setTableType(event.target.value);
+    }
+
+    @action.bound
+    onConfirmClick(order_info) {
+        this.general_store.props.setOrderId(order_info.id);
+        this.general_store.redirectTo('orders', { nav: { location: 'buy_sell' } });
+    }
+
+    @action.bound
+    setApiErrorMessage(api_error_message) {
+        this.api_error_message = api_error_message;
+    }
+
+    @action.bound
+    setContactInfo(contact_info) {
+        this.contact_info = contact_info;
+    }
+
+    @action.bound
+    setErrorMessage(error_message) {
+        this.error_message = error_message;
+    }
+
+    @action.bound
+    setFormProps(props) {
+        this.form_props = props;
+    }
+
+    @action.bound
+    setHasMoreItemsToLoad(has_more_items_to_load) {
+        this.has_more_items_to_load = has_more_items_to_load;
+    }
+
+    @action.bound
+    setIsLoading(is_loading) {
+        this.is_loading = is_loading;
+    }
+
+    @action.bound
+    setIsSubmitDisabled(is_submit_disabled) {
+        this.is_submit_disabled = is_submit_disabled;
+    }
+
+    @action.bound
+    setItemOffset(item_offset) {
+        this.item_offset = item_offset;
+    }
+
+    @action.bound
+    setItems(items) {
+        this.items = items;
+    }
+
+    @action.bound
+    setPaymentInfo(payment_info) {
+        this.payment_info = payment_info;
+    }
+
+    @action.bound
+    setInitialReceiveAmount() {
+        this.receive_amount = getRoundedNumber(
+            this.advert.min_order_amount_limit * this.advert.price,
+            this.advert.local_currency
+        );
```

Comment:
Can be a `computed` value rather than a function?

Reply:
It's being set in `/packages/p2p/src/components/buy-sell/buy-sell-form.jsx` though.

Reply:
This value is the initial value as the form opens, after that `receive_amount` is set with changing `input_amount` value. If I need to use computed with that I'll have to set the `input_amount` as it changes. Which would be better in this case? Leaving `receive_amount` as is or creating an observable for `input_amount`?